controllers/storagecluster/cephcluster.go

```diff
@@ -110,6 +130,37 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 		return reconcile.Result{}, fmt.Errorf("'StorageDeviceSets' should not be initialized in an external CephCluster")
 	}
 
+	// TODO: remove the fix in the next release 4.19
+	// as it is needed only for the upgrade to 4.18
+	if !sc.Spec.ExternalStorage.Enable {
+		// delete the osd prepare jobs, which are completed and have duplicated placements
+		jobName := "rook-ceph-osd-prepare"
+		listOpts := &client.ListOptions{
+			Namespace: sc.Namespace,
+			LabelSelector: labels.SelectorFromSet(map[string]string{
+				"app": jobName,
+			}),
+		}
+		jobs := &batchv1.JobList{}
+		// list all the prepare jobs
+		err := r.Client.List(r.ctx, jobs, listOpts)
+		if err != nil {
+			r.Log.Error(err, "failed to list prepare jobs")
+			return reconcile.Result{}, err
+		}
+
+		// filter out the completed jobs
+		for _, job := range jobs.Items {
+			if job.Status.Succeeded == 1 {
+				err := r.deletePlacementJobsWithDuplicateTolerations(job)
```

Comment:
```suggestion
				err := r.deleteJobsWithDuplicateTolerations(job)
```
Don't think we need `placement` in that name.